deployments-npm-package/__tests__/test_package.ts

```diff
@@ -0,0 +1,113 @@
+import path from 'path';
+import { execSync } from 'child_process';
+import fs from 'fs/promises';
+
+const pathToArtifacts = `${__dirname}/../../deployments/artifacts`;
+/**
+ * We read the artifacts directory and get read all the json artifacts, from there we extract the contracts
+ * inside of the base or chain key and return them as an array.
+ *
+ * @returns {Promise<{ [key: string]: string[] }>}
+ */
+const getContracts = async () => {
+  const contractsPerNetwork: { [key: string]: string[] } = {};
+  try {
+    const artifactItems = await fs.readdir(pathToArtifacts);
+
+    for (const item of artifactItems) {
+      const artifactPath = path.join(pathToArtifacts, item);
+      try {
+        const stat = await fs.stat(artifactPath);
+        if (stat.isDirectory() && !item.startsWith('.')) {
+          const files = await fs.readdir(artifactPath);
+
+          for (const file of files) {
+            const filePath = path.join(artifactPath, file);
+            const data = await fs.readFile(filePath, 'utf8');
+            const json = JSON.parse(data);
+
+            const baseContracts = json.base ? Object.keys(json.base) : [];
+            const chainContracts = json.chain ? Object.keys(json.chain) : [];
+
+            const contracts = [...baseContracts, ...chainContracts].filter(
+              (key) => key !== 'chain_id',
+            );
+
+            if (!contractsPerNetwork[item]) {
+              contractsPerNetwork[item] = contracts;
+            } else {
+              contractsPerNetwork[item] = [...contractsPerNetwork[item], ...contracts];
+            }
```

Comment:
We could move the Set initialization here. I believe this will work better and be more understandable:
```js
const contracts = new Set([...baseContracts, ...chainContracts]);
contracts.delete('chain_id');

if (!contractsPerNetwork[item]) {
  contractsPerNetwork[item] = [...contracts];
} else {
  contractsPerNetwork[item] = [...contractsPerNetwork[item], ...contracts];
}
```

By having this, you can remove lines 51-53 below.